File: spark16/src/main/java/scalassssss/CartesianOperator.scala
package scalassssss

import org.apache.spark.{SparkConf, SparkContext}
import java.util.Arrays
/**
  * Created by 小省 on 2016/7/7.
  */
/*
笛卡尔积
 */
object CartesianOperator {
  def main(args: Array[String]) {

    val conf=new SparkConf().setAppName("CartesianOperator").setMaster("local[4]")
    val sc =new SparkContext(conf)

    val arr1=Array("T恤衫","夹克","皮大衣","衬衫","毛衣")
    val arr2=Array("西裤","内裤","铅笔裤","皮裤","牛仔裤")

    val lines1=sc.parallelize(arr1)
    val lines2=sc.parallelize(arr2)

    lines1.cartesian(lines2).foreach(println(_))
  }

}

File: spark16/src/main/java/javasssss/CartesianOperator.java
package javasssss;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import scala.Tuple2;

import java.util.Arrays;
import java.util.List;

/**
 * Created by 小省  on 2016/6/15.
 */
public class CartesianOperator {
	public static void main(String[] args) {
		SparkConf conf =new SparkConf().setAppName("CartesianOperator").setMaster("local");
		JavaSparkContext sc =new JavaSparkContext(conf);

		List<String> clothes = Arrays.asList("T恤衫","夹克","皮大衣","衬衫","毛衣");
		List<String> trousers = Arrays.asList("西裤","内裤","铅笔裤","皮裤","牛仔裤");
		JavaRDD<String> clothesRDD = sc.parallelize(clothes);
		JavaRDD<String> trousersRDD = sc.parallelize(trousers);

		
		trousersRDD.count();
		JavaPairRDD<String,String> result= clothesRDD.cartesian(trousersRDD);

		for (Tuple2<String,String> as:result.collect()){
			System.out.println(as);
		}

		sc.close();



	}
}
